internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -34,62 +34,223 @@
 package metadata_manager
 
 import (
+	"encoding/json"
+	"path/filepath"
+	"strconv"
+	"time"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
 	"github.com/Azure/azure-storage-fuse/v2/internal/dcache"
 )
 
 // BlobMetadataManager is the implementation of MetadataManager interface
 type BlobMetadataManager struct {
-	cacheDir string
+	cacheDir         string
+	storageCallbacks dcache.StorageCallbacks
 }
 
 // NewMetadataManager creates a new implementation of the MetadataManager interface
-func NewMetadataManager(cacheDir string) (MetadataManager, error) {
+func NewMetadataManager(cacheDir string) (*BlobMetadataManager, error) {
 	return &BlobMetadataManager{
 		cacheDir: cacheDir,
 	}, nil
 }
 
 // CreateFileInit creates the initial metadata for a file
 func (m *BlobMetadataManager) CreateFileInit(filePath string, fileMetadata *dcache.FileMetadata) error {
```

Comment:
what's the reason we take FileMetadata and not []byte, like the heartbeat and other APIs?
Unless we have a need to interpret that inside the metadata manager (which it should not) let's be consistent and use []byte for file APIs too.

Reply:
Also, thinking more about it, this should return the etag of the newly created blob, so that caller can then pass that to CreateFileFinalize() to be sure it's not finalizing a file that someone else created. This would be useful to handle cases where some node creates the initial file, then maybe goes silent for a while and some other node takes over and creates a fresh init file and then the original node came up and wants to finalize the file.

For now you can add a TODO and we can address it later.